library/x509_create.c

```diff
@@ -307,8 +309,19 @@ int mbedtls_x509_write_sig(unsigned char **p, unsigned char *start,
 
     // Write OID
     //
-    MBEDTLS_ASN1_CHK_ADD(len, mbedtls_asn1_write_algorithm_identifier(p, start, oid,
-                                                                      oid_len, 0));
+    if (pk_alg == MBEDTLS_PK_ECDSA) {
```

Comment:
I'm a bit disappointed that the logic “`write_null_par = (pk_alg is RSA)`” is repeated many times, but I can't find a neat way to avoid having to repeat it. It would be nice to decide at the same time we look up the OID, but the OID module isn't really set up to convey this information.